### ISSUE TITLE
Feature Request: Support kubernetes.io/ingress.class annotation

### ISSUE THREAD

**Issue by rezonant** [rancher/rancher#6344]
**Rancher Version:** 1.1.4
**Reproduction**
1. add `kubernetes.io/ingress-class=foobar` annotation to a k8s ingress resource. _Result_: Rancher's k8s integration sets up a rancher LB for the ingress. _Expected_: Rancher's k8s integration should ignore it. 
2. stop service `rancher-ingress-controller` from rancher UI
3. click Catalog menu item
_Result_: Rancher shows the Setting Up Kubernetes screen, showing 8/9 services as ready (obviously the last service is rancher-ingress-controller). This condition will never resolve. The only way to restore the k8s integration is to restart `rancher-ingress-controller`. _Expected_: Previous issue #5999 seems to indicate that this works

**Notes:**
From what I gather, this annotation is fairly new, or at least was only implemented in the nginx ingress controller about a week ago (https://github.com/nginxinc/kubernetes-ingress/issues/49). I realize that it should be possible to "upgrade" `rancher-ingress-controller` to a no-op docker image, but this is a pretty big hack, and its not obvious that it has no deleterious side effects. Also, this would make Rancher-controlled ingresses unusable in the cluster, whereas the standard annotation would allow multiple ingress controllers to co-exist in a Rancher-managed cluster.

**Rationale:**
Use case: A user should be able to use traefik, the nginx ingress controller, or a custom solution instead of or in addition to the built-in Rancher ingress controller. 

This was previously mentioned in #5999, and it was recommended to disable `rancher-ingress-controller` service, however this causes the Rancher UI to revert to the Kubernetes Setup user interface. This causes most of the Rancher k8s integration to no longer function, including the Services, Pods, Replication Controllers and kubectl pages.
**UPDATE**: Rancher 1.2.0 allows you to disable this service without affecting the Kubernetes UI. However, this causes the stack to become degraded (yellow)

**Comment by LoicMahieu** [comment]
I am also in this issue.

I would like to use my Kubernetes cluster for multiple apps, which have their own `Ingress` with only host based rules.
My first setup was to disable/remove `rancher-ingress-controller` and install traefik load balancer which works pretty well!
Unfortunatly, like you said, the Rancher UI does not like you disable some part of the stack.

We tried to fork the official catalog to remove the `rancher-ingress-controller` from the templates. We disabled the official catalog and added a new called `library`.
It worked well... A few minor issue with `git clone`... But it worked... But I am not confortable with this technic. We had to maintain this fork in sync with official etc...

**Comment by rezonant** [comment]
Rancher 1.2.0 came out and I naturally re-evaluated the options when it comes to handling ingress and automated TLS.

@LoicMahieu In the new release I can disable the ranger ingress controller service without causing the Kubernetes UI to stop. I think it bumps the stack to warning (yellow) though. 

I have also experimented with replacing the service with a docker no-op image which might get your stack green, but iirc I ran into a few problems with that approach. 

Traefik is offered in the Kubernetes catalog as an infrastructure stack but it launches on a different port and is doing nothing to stop Rancher's built in ingress controller from taking over.

Unfortunately, due to the lack of this feature and the fact that Rancher's built in ingress controller does not seem to support multiple host-based ingresses yet, I am still using the custom solution I developed for Rancher 1.1 series using metadata on services instead of proper ingress resources :-\

**Comment by deniseschannon** [comment]
@rezonant I'm not sure what you mean by "multiple host-based ingresses", but we do support host based routing and I would assume it could be for multiple targes. 

http://docs.rancher.com/rancher/v1.2/en/kubernetes/ingress/#setting-up-a-ingress-resource-with-host-based-routing

**Comment by rezonant** [comment]
Newer versions of Kubernetes allow you to create multiple Ingress resources, each of which may vary only by hostname. The intention of this is to allow the infrastructure layer to construct a _single_ load balancer which can connect a single ingress port to Kubernetes services via the ingress services which they are associated with.

**Comment by rezonant** [comment]
Adding two ingress resources causes rancher's LB controller to do this:

![image](https://cloud.githubusercontent.com/assets/409140/21284273/4aa8a7ba-c3e5-11e6-8f02-9bf5851ca1cb.png)

**Comment by joshwget** [comment]
It will be possible to disable the Rancher ingress controller in an upcoming version of Rancher (#5999). If you're looking to use the nginx ingress controller exclusively this would solve your issue.

**Comment by mishak87** [comment]
IMO it should ignore `kubernetes.io/ingress-class` that is not empty or equal to rancher.
Is there something preventing fix like this get into 1.5.* releases?

**Comment by deniseschannon** [comment]
Available in the`v1.7.4-rancher1-etcdv2` template with `rancher/lb-service-rancher:v0.7.10`.

**Comment by galal-hussein** [comment]
Tested on Rancher v1.6.8-rc4
Kubernetes: rancher/k8s:v1.7.4-rancher1
rancher ingress controller: rancher/lb-service-rancher:v0.7.10

Tested creating two ingresses one with the `kubernetes.io/ingress.class` and one without

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: cafe-ingress-nginx
  annotations:
    kubernetes.io/ingress.class: "nginx"
spec:
  rules:
  - http:
      paths:
      - path: /tea
        backend:
          serviceName: tea-svc
          servicePort: 80
      - path: /coffee
        backend:
          serviceName: coffee-svc
          servicePort: 80
```
I can verify that when creating an ingress without annotation a lb container service is created, and when using the annotation an ingress resource is created and routed throguh the ingress controller
